<p align="center">
    <a href="https://latch.bio/">
        <img src="images/latch_logo.png" width=37.5%>
    </a>
    <a href="https://latch.bio/">
        <img src="images/iqtree_logo.svg" width=18%>
    </a>
</p>

<p align="center">
    <a href="https://console.latch.bio/explore/60254/info">
    <span style="font-size:larger;">Click here to see the workflow!</span></br>
    </a>
    </br></br>
    <a href="https://github.com/hannahle/alignment/graphs/contributors" alt="Contributors">
        <img src="https://img.shields.io/github/contributors/jlsteenwyk/latch_wf_infer_phylogeny">
    </a>
    <a href="https://twitter.com/intent/follow?screen_name=jlsteenwyk" alt="Author Twitter">
        <img src="https://img.shields.io/twitter/follow/jlsteenwyk?style=social&logo=twitter"
            alt="follow on Twitter">
    </a>
    <a href="https://twitter.com/intent/follow?screen_name=matteobolner" alt="Author Twitter">
        <img src="https://img.shields.io/twitter/follow/matteobolner?style=social&logo=twitter"
            alt="follow on Twitter">
    </a>
</p>

</br>

# THIS WORKFLOW IS STILL IN DEVELOPMENT AND IS CURRENTLY NOT AVAILABLE

# IQTREE, efficient phylogenomic/phylogenetic inference
## About
IQTREE infers phylogenies from multiple sequence alignments.

<br />

If you found IQTREE useful, please cite *IQ-TREE 2: New models and efficient
methods for phylogenetic inference in the genomic era*. Minh et al. 2020,
Molecular Biology and Evolution. doi:
[10.1093/molbev/msaa015](https://doi.org/10.1093/molbev/msaa015).

<br />

To run IQTREE, user's must provide a minimum a multiple sequence
alignment. This workflow conducts automated substitution matrix
model testing. Specify output prefixes and the number of ultrafast
bootstrap approximations (or UFBoot) to run.
